Tests/base/NSPredicate/nsnull.m

```diff
@@ -0,0 +1,34 @@
+#import "ObjectTesting.h"
+#import "Foundation/NSAutoreleasePool.h"
+#import "Foundation/NSPredicate.h"
+#import "Foundation/NSArray.h"
+#import "Foundation/NSNull.h"
+
+int main(void) {
+    NSAutoreleasePool *arp = [NSAutoreleasePool new];
+	NSArray *array, *filtered;
+	NSPredicate *predicate;
+
+    // Basic filtering with NSPredicate
+    array = @[@{@"key": @"value1"}, @{@"key": @"value2"}, [NSNull null]];
```

Comment:
This notation will break with gcc based compilation.

Reply:
Fixed

Reply:
Seems like this is also not allowed:

```sh
nsnull.m: In function ‘main’:
nsnull.m:21:45: error: array subscript is not an integer
   21 |     PASS(filtered.count == 1 && [filtered[0][@"key"] isEqualToString:@"value2"],
   ```